### PR TITLE
Don't unnecessarily download or merge when work was previously done.

### PR DESCRIPTION
**Pull request by nickmshelley**

### CODE REVIEW

File: LDSContent/ContentController.swift [hunk 1]
```diff
@@ -74,24 +74,27 @@ public class ContentController {
     /// Checks the server for the latest catalog version and installs it if newer than the currently
     /// installed catalog (or if there is no catalog installed).
     public func updateCatalog(secureCatalogs secureCatalogs: [(name: String, baseURL: NSURL)]? = nil, progress: (amount: Float) -> Void = { _ in }, completion: (UpdateAndMergeCatalogResult) -> Void) {
-        session.updateDefaultCatalog { result in
+        if let secureCatalogs = secureCatalogs {
+            // Delete secure catalogs we no longer have access to
+            let catalogsToDelete = self.contentInventory.installedCatalogs().filter { installedCatalog in !installedCatalog.isDefault() && !secureCatalogs.contains({ secureCatalog in secureCatalog.name == installedCatalog.name })}.map { $0.name }
```
Comment (hiltonc) [inline LDSContent/ContentController.swift]: Don't need self.

File: LDSContent/DownloadCatalogOperation.swift [hunk 2]
```diff
@@ -31,27 +31,43 @@ enum DownloadCatalogOperationError: ErrorType {
...
     let session: Session
+    let baseURL: NSURL
+    let destination: (version: Int) -> NSURL
     let progress: (amount: Float) -> Void
     let tempDirectoryURL: NSURL
-    let catalogName: String
-    let baseURL: NSURL
     
-    init(session: Session, catalogName: String, baseURL: NSURL, progress: (amount: Float) -> Void, completion: (DownloadCatalogResult) -> Void) {
+    init(session: Session, catalogName: String, baseURL: NSURL, destination: (version: Int) -> NSURL, progress: (amount: Float) -> Void, completion: (DownloadCatalogResult) -> Void) {
```
Comment (hiltonc) [inline LDSContent/DownloadCatalogOperation.swift]: That's a weird closure.

**Comment by hiltonc** [comment]
👍 But please check the demo app to make sure it works OK.

File: LDSContent/DownloadCatalogOperation.swift (hunk 2 above)
Comment (nickmshelley) [inline LDSContent/DownloadCatalogOperation.swift]: Was this an observation or an action item? If the latter, I can't figure out what... :)

**Comment by nickmshelley** [comment]
Yep demo works great.

File: LDSContent/ContentController.swift [hunk 3]
```diff
@@ -74,24 +74,27 @@ public class ContentController {
...
-                
+            case let .Success(version):
                 do {
                     try self.contentInventory.addOrUpdateCatalog(ContentController.defaultCatalogName, url: nil, version: version)
                 } catch let error as NSError {
                     completion(.Error(errors: [error]))
                     return
                 }
-                
+                fallthrough
```
Comment (stephanheilner) [inline LDSContent/ContentController.swift]: I've never seen `fallthrough`, what does it do?

File: LDSContent/DownloadCatalogOperation.swift (hunk 2 above)
Comment (hiltonc) [inline LDSContent/DownloadCatalogOperation.swift]: You pick :)

Comment (nickmshelley) [inline LDSContent/DownloadCatalogOperation.swift]: Well I think it's necessary and I don't know how to make it less "weird," so... I pick observation!

File: LDSContent/ContentController.swift (hunk 3 above)
Comment (nickmshelley) [inline LDSContent/ContentController.swift]: It falls through to the next case.

**Comment by nickmshelley** [comment]
Is this good to merge?

**Comment by hiltonc** [comment]
See previous 👍

**Comment by nickmshelley** [comment]
But that was before my last commit.

**Comment by hiltonc** [comment]
:+1:

**Pull request merged**